compiler/testData/codegen/bytecodeText/inlineClasses/resultApiDoesntCallSpecializedEquals.kt

```diff
@@ -0,0 +1,13 @@
+// WITH_COROUTINES
+// FILE: test.kt
+fun test() {
+    val result = Result.success("yes!")
+    val other = Result.success("nope")
+    if (result == other) println("==")
+    if (result != other) println("!=")
+    if (result.equals(other)) println("equals")
+    if (!result.equals(other)) println("!equals")
```

Comment:
Judging by [KT-33722](https://youtrack.jetbrains.com/issue/KT-33722), these two lines should fail. Probably they don't because we only call `Result.equals-impl0` if API version >= 1.4. So let's add `!API_VERSION: LATEST` to this test, so that it tests something useful. :)

But also, I don't really think the explicit `equals` call is such critical behavior that we should check it at this point in a codegen test. Having it reported as KT-33722 is probably enough. So I'd probably just remove these two lines from the test.

Reply:
Right, I don't think specializing explicit `equals` calls is very relevant either. I've removed this part and also got rid of `testResultApiEqualsCallDoesntBox`.

Apart from that, I've changed the test so that it explicitly runs with an old API version (and doesn't use `equals-impl0`), while `testResultApiEquals` runs with the latest API version (and doesn't box).